### PR TITLE
Publish to npm

### PR DESCRIPTION
**Pull request by jasonkarns**
Updates to package.json for publishing to npm.

- [ ] author/contributor details good?
- [ ] compatibility with bpkg https://github.com/bpkg/bpkg/issues/17
- [ ] keywords?

The referenced issue from bpkg indicates that they intend to move off of package.json, or otherwise find a way to make bpkg's use compatible with npm. Per that issue, I don't know if that has happened or not. But as this PR stands at present, I _believe_ it will break compatibility with bpkg due to the conflicting semantics of `scripts`.

For now, my intention is to first get the metadata correct (authorship info, keywords, etc) and publish the currently tagged versions to the npm registry. This can occur without having the package.json merged to master, so we don't necessarily have to resolve the bpkg compatibility issue _immediately_.

- [ ] I have reviewed the [Contributor Guidelines][contributor].
- [ ] I have reviewed the [Code of Conduct][coc] and agree to abide by it

[contributor]: https://github.com/bats-core/bats-core/blob/master/docs/CONTRIBUTING.md
[coc]:         https://github.com/bats-core/bats-core/blob/master/docs/CODE_OF_CONDUCT.md

### CODE REVIEW

File: package.json
Comment (mbland) [inline package.json]: Any reason why this is set back from `v1.0.1`?

File: package.json
Comment (mbland) [inline package.json]: Though Sam's the original author, of course, I wonder if it wouldn't be more accurate to list https://github.com/bats-core as the author and Sam as a contributor.

**Comment by jasonkarns** [comment]
I should note a couple more things in the package:
- files are included explicitly in the tarball via the `files` stanza. This way the published tarball only includes necessary files.
- bins are specified via `directories.bin` (which is mutually exclusive with the `bin` stanza)
- contributor metadata was taken from github profiles
- the `scripts` I expect will eventually cover publishing to npm, docker, and version bumping. I would like to see these steps follow the https://github.com/github/scripts-to-rule-them-all pattern, but is immaterial to getting 1.0.0 and 1.0.1 published.

File: package.json
Comment (mbland) [inline package.json]: Is this for bpkg compatibility? The `github:bats-core/bats-core` ID meets the current package.json format: https://docs.npmjs.com/files/package.json#repository

File: package.json
Comment (mbland) [inline package.json]: Would an AUTHORS file be more desirable? Just got the idea from https://docs.npmjs.com/files/package.json#default-values.

File: package.json
Comment (mbland) [inline package.json]: So `directories.bin` obviates the need for the `bin` parameter? https://docs.npmjs.com/files/package.json#directories

File: package.json
Comment (jasonkarns) [reply package.json]: I'm going to publish both 1.0.0 and 1.0.1 versions once the rest of the metadata is approved, so (for this PR which _won't_ be merged) I'm keeping it at the old version. That way if I forget to bump it before attempting to publish, it'll error (existing version).

File: package.json
Comment (jasonkarns) [reply package.json]: This is the format that was generated by `npm init`. I'm cool switching to the terse form.

File: package.json
Comment (jasonkarns) [reply package.json]: Yep

File: package.json
Comment (jasonkarns) [reply package.json]: Yeah, I'm cool with that. Generally I try to minimize files in repo root. But since AUTHORS would (presumably) support non-node/npm tools, I'm 👍

File: package.json
Comment (jasonkarns) [reply package.json]: Eh, the definitions I typically adhere to:

Author: The person/s or organization that created the project
Owner: The person/s who has administrative ownership over the organization or repository
Maintainers: Contributors who are responsible for driving the vision and managing the organizational aspects of the project.
Contributors: Everyone who has contributed something back to the project.

https://opensource.guide/how-to-contribute/#anatomy-of-an-open-source-project

Obviously, there are many (conflicting) definitions and there's nothing that states github's version as authoritative. However, by switching to AUTHORS file for contributors, that _kinda_ muddies the distinction (which could be seen as a good thing?)

I'd prefer to keep Sam as author in package.json, and move contributors to AUTHORS file. google seems to distinguish AUTHORS file from CONTRIBUTORS file: https://opensource.google.com/docs/releasing/authors/

I wouldn't mind re-addressing this if/when bats-core gets the official nod as maintained successor to bats.

File: package.json
Comment (jasonkarns) [reply package.json]: (FYI, I intend to publish from the commits actually tagged per version)

File: package.json
Comment (mbland) [reply package.json]: As far as an official nod goes, it'd be lovely if that would happen, and I'd welcome and celebrate it, as that's the order of things I'd prefer (short of being able to contribute to the original project, obviating the need for a fork)—but I've stopped holding my breath.

Comment (jasonkarns) [reply package.json]: Same. But until that happens and/or we significantly change the code, I'd prefer to just maintain status quo with him as author?

Comment (mbland) [reply package.json]: It's arguable we have, but it's not a blocker, just something about which I was curious to hear others' thoughts.

**Comment by jasonkarns** [comment]
Throw me your npm registry usernames and I'll add you as maintainers.

**Comment by mbland** [comment]
@jasonkarns My npm username is @mbland as well.

**Comment by jasonkarns** [comment]
1.0.0 and 1.0.1 have been published to npm. Merging this for now so it's ready for future releases.

**Comment by mbland** [comment]
@jasonkarns Just pushed [1.0.2](https://github.com/bats-core/bats-core/releases/tag/v1.0.2). Mind pushing it to npm, or giving me collaborator permission, whatever is easiest in the short term?

**Comment by jasonkarns** [comment]
@mbland you should already have maintainer access. (sent invite last week or so?). ping me if that's not the case.

related, I would like to see how interested people might be in using `npm version (major|minor|patch)` to do the version bumping? that way it can do the version bump, update all the necessary files automatically, tag/commit, git-push and publish. It would make node a requirement for us maintainers, but as long as we're all publishing to npm anyway... /shrug  (or we could have it published by CI? though it's still nice to have all the version bumping automated)

**Comment by mbland** [comment]
Oh! That's weird. I didn't get an invite email, and I still only see your badge on https://www.npmjs.com/package/bats. But I could click through on https://www.npmjs.com/package/bats/access, see myself listed, and then `npm publish` v1.0.2.

I'm leaning against using `npm version` to do the bumping, as I don't see it as much work and don't mind doing it manually; but if you want to figure out how to get it all to work, feel free to try. One thing I like to do that I don't see in https://docs.npmjs.com/cli/version is to do annotated tags, where I've been posting the release notes as well.

On a related note, I'm wondering if it might be worth moving the "Version history" from the README into its own CHANGES document or some such.

**Comment by jasonkarns** [comment]
> annotated tags

I believe it does annotated tags by default, but it just uses the version number by default. I'll play around and see if we can customize that.



> "Version history" from the README into its own CHANGES document

Strong agree. For other things, I also use `hub` to publish release notes to each github Release. Would be nice to have all that connected. (Release including link to changelog, etc)

**Review by mizo7777 — APPROVED** [review]

**Pull request merged**
